table/index_builder.cc

```diff
@@ -24,21 +25,22 @@ namespace rocksdb {
 IndexBuilder* IndexBuilder::CreateIndexBuilder(
     BlockBasedTableOptions::IndexType index_type,
     const InternalKeyComparator* comparator,
+    const SliceTransform* slice_transform,
```

Comment:
prefix_extractor comes from _ioptions.prefix_extractor which is of type SliceTransform.

slice_transform comes from this->internal_prefix_transform which is of type InternalKeySliceTransform initialized by prefix_extractor's transform member. slice_transform is used only by the hash index builder.

previously I was reusing InternalKeySliceTransform for also partitioned filter builder which turns out to be wrong. So here we distinguish between the two, use slice_transform for hash (as it was before) and use _ioptions.prefix_extractor for partitioned filter.

I understand the naming is confusing here. I am going to ahead doing this renaming:
slice_transform => int_key_slice_transform
prefox_extractor => slice_transform and also use the explicit type of InternalKeySliceTransform for it.